# Rosalind
## About
The repository contains solutions to problems posted on rosalind.info.
Each exercise is provided with example dataset.   
## Language
Scripts were written in Python 3.7.
## Running a script
In order to run a script pass a path to corresponding dataset as an argument.
## Bioinformatics Stronghold
<center>

ID | Title
---|------
DNA | [Counting DNA Nucleotides](./BioinformaticsStronghold/dna/dna.py)
RNA | [Transcribing DNA into RNA](./BioinformaticsStronghold/rna/rna.py)
REVC | [Complementing a Strand of DNA](./BioinformaticsStronghold/revc/revc.py)
GC | [Computing GC Content](./BioinformaticsStronghold/gc/gc.py)
HAMM | [Counting Point Mutations](./BioinformaticsStronghold/hamm/hamm.py)
PROT | [Translating RNA into Protein](./BioinformaticsStronghold/prot/prot.py)
SUBS | [Finding a Motif in DNA](./BioinformaticsStronghold/subs/subs.py)
ORF | [Open Reading Frames](./BioinformaticsStronghold/orf/orf.py)
PERM | [Enumerating Gene Orders](./BioinformaticsStronghold/perm/perm.py)
PRTM | [Calculating Protein Mass](./BioinformaticsStronghold/prtm/prtm.py)
FIB | [Rabbits and Recurrence Relations](./BioinformaticsStronghold/fib/fib.py)
CONS | [Consensus and Profile](./BioinformaticsStronghold/cons/cons.py)
<!---    
FIB | Rabbits and Recurrence Relations     
IPRB | Mendel's First Law  
FIBD | Mortal Fibonacci Rabbits  
GRPH | Overlap Graphs  
IEV | Calculating Expected Offspring  
LCSM | Finding a Shared Motif  
LIA | Independent Alleles  
MPRT | Finding a Protein Motif  
MRNA | Inferring mRNA from Protein 
REVP | Locating Restriction Sites  
SPLC | RNA Splicing  
LEXF | Enumerating k-mers Lexicographically  
LGIS | Longest Increasing Subsequence  
LONG | Genome Assembly as Shortest Superstring  
PMCH | Perfect Matchings and RNA Secondary Structures  
PPER | Partial Permutations  
PROB | Introduction to Random Strings  
SIGN | Enumerating Oriented Gene Orderings  
SSEQ | Finding a Spliced Motif  
TRAN | Transitions and Transversions  
TREE | Completing a Tree  
CAT | Catalan Numbers and RNA Secondary Structures  
CORR | Error Correction in Reads  
INOD | Counting Phylogenetic Ancestors  
KMER | k-Mer Composition  
KMP | Speeding Up Motif Finding  
LCSQ | Finding a Shared Spliced Motif  
LEXV | Ordering Strings of Varying Length Lexicographically  
MMCH | Maximum Matchings and RNA Secondary Structures  
PDST | Creating a Distance Matrix  
REAR | Reversal Distance  
RSTR | Matching Random Motifs  
SSET | Counting Subsets  
ASPC | Introduction to Alternative Splicing  
EDIT | Edit Distance  
EVAL | Expected Number of Restriction Sites  
MOTZ | Motzkin Numbers and RNA Secondary Structures  
NWCK | Distances in Trees  
SCSP | Interleaving Two Motifs  
SETO | Introduction to Set Operations  
SORT | Sorting by Reversals  
SPEC | Inferring Protein from Spectrum  
TRIE | Introduction to Pattern Matching  
CONV | Comparing Spectra with the Spectral Convolution  
CTBL | Creating a Character Table  
DBRU | Constructing a De Bruijn Graph  
EDTA | Edit Distance Alignment  
FULL | Inferring Peptide from Full Spectrum  
INDC | Independent Segregation of Chromosomes  
ITWV | Finding Disjoint Motifs in a Gene  
LREP | Finding the Longest Multiple Repeat  
NKEW | Newick Format with Edge Weights  
RNAS | Wobble Bonding and RNA Secondary Structures  
AFRQ | Counting Disease Carriers  
CTEA | Counting Optimal Alignments  
GLOB | Global Alignment with Scoring Matrix  
SGRA | Using the Spectrum Graph to Infer Peptides  
GCON | Global Alignment with Constant Gap Penalty  
LOCA | Local Alignment with Scoring Matrix  
MGAP | Maximizing the Gap Symbols of an Optimal Alignment  
MULT | Multiple Alignment  
SEXL | Sex-Linked Inheritance  
GAFF | Global Alignment with Scoring Matrix and Affine Gap Penalty
OAP | Overlap Alignment
SIMS | Finding a Motif with Modifications
SMGB | Semiglobal Alignment
LAFF | Local Alignment with Affine Gap Penalty
OSYM | Isolating Symbols in Alignments
MEND | Inferring Genotype from a Pedigree
CSTR | Creating a Character Table from Genetic Strings
WFMD | The Wright-Fisher Model of Genetic Drift
EBIN | Wright-Fisher's Expected Behavior
FOUN | The Founder Effect and Genetic Drift
CUNR | Counting Unrooted Binary Trees
ROOT | Counting Rooted Binary Trees
PCOV | Genome Assembly with Perfect Coverage
ASMQ | Assessing Assembly Quality with N50 and N75
GASM | Genome Assembly Using Reads
---> 
</center>

## Algorithmic Heights
<center>

ID | Title
---|------
FIBO | [Fibonacci Numbers](./AlgorithmicHeights/fibo/fibo.py)
INS | [Insertion Sort](./AlgorithmicHeights/ins/ins.py)
MER | [Merge Two Sorted Arrays](./AlgorithmicHeights/mer/mer.py)
BINS | [Binary Search](./AlgorithmicHeights/bins/bins.py)
MS | [Merge Sort](./AlgorithmicHeights/ms/ms.py)
MAJ | [Majority Element](./AlgorithmicHeights/maj/maj.py)
HEA | [Building a Heap](./AlgorithmicHeights/hea/hea.py)
HS | [Heap Sort](./AlgorithmicHeights/hs/hs.py)
INV | [Counting Inversions](./AlgorithmicHeights/inv/inv.py)
2SUM | [2SUM](./AlgorithmicHeights/2sum/2sum.py)
</center>

## Bioinformatics Armory
<center>

ID | Title
---|------
INI | [Introduction to the Bioinformatics Armory](./BioinformaticsArmory/ini/ini.py)
</center>

## Python Village
<center>

ID | Title
---|------
INI1 | [Installing Python](./PythonVillage/ini1/ini1.py)
INI2 | [Variables and Some Arithmetic](./PythonVillage/ini2/ini2.py)
INI3 | [Strings and Lists](./PythonVillage/ini3/ini3.py)
INI4 | [Conditions and Loops](./PythonVillage/ini4/ini4.py)
INI5 | [Working with Files](./PythonVillage/ini5/ini5.py)
INI6 | [Dictionaries](./PythonVillage/ini6/ini6.py)
</center>